### PR TITLE
(PC-6343) Autofocus des inputs

### PR DESCRIPTION
**Pull request by BenStirrup**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6343

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
      (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

### CODE REVIEW

File: src/ui/components/inputs/BaseTextInput.tsx [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
+
+    /** BUG FIX for :
+     * text input of type password (with secureTextEntry set to True) has not the
+     * correct fontFamily despite the style
+     */
+    inputRef.current.setNativeProps({ style: { fontFamily: 'Montserrat-Regular' } })
+
+    /** FEATURE HACK for : autoFocus with keyboard display on Android
+     * How :
+     * 1. Do NOT set the autoFocus prop on the native TextInput
```
Comment (blixit) [inline src/ui/components/inputs/BaseTextInput.tsx:31]: on peut comprendre que t'as du le faire, mais la raison n'est pas claire

File: src/ui/components/inputs/BaseTextInput.tsx [hunk 2]
```diff
@@ -1,32 +1,49 @@
...
+     * 2. Trigger the ref.focus() function with a setTimeout on mount.
```
Comment (blixit) [inline src/ui/components/inputs/BaseTextInput.tsx:32]: idem, tu peux expliciter la raison du délai, c'est cette raison qui va permettre à un prochain dev de ne pas casser le code

**Review by emmajmn — APPROVED** [review]

File: src/ui/components/inputs/BaseTextInput.tsx (hunk 2 above)
Comment (BenStirrup) [reply src/ui/components/inputs/BaseTextInput.tsx]: done !

File: src/ui/components/inputs/BaseTextInput.tsx (hunk 1 above)
Comment (BenStirrup) [reply src/ui/components/inputs/BaseTextInput.tsx]: done !

**Review by blixit — APPROVED** [review]

**Pull request merged**